src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncMethodBuilderMemberCollection.cs

```diff
@@ -422,6 +424,17 @@ private static NamedTypeSymbol ValidateBuilderType(SyntheticBoundNodeFactory F,
                     !property.IsStatic &&
                     (property.ParameterCount == 0))
                 {
+                    HashSet<DiagnosticInfo> useSiteDiagnostics = null;
+                    Conversion conversion = F.Compilation.Conversions.ClassifyConversionFromType(property.Type, returnType, ref useSiteDiagnostics);
+                    if (conversion.Kind == ConversionKind.NoConversion)
+                    {
+                        var badTaskProperty = new CSDiagnostic(
+                            new CSDiagnosticInfo(ErrorCode.ERR_BadAsyncMethodBuilderTaskProperty, builderType, returnType, property.Type),
+                            F.Syntax.Location);
+                        F.Diagnostics.Add(badTaskProperty);
+                        return null;
```

Comment:
For other `AsyncMethodBuilder` methods, we're simply reporting `CS0656: Missing compiler required member`. That's not particularly informative but it is consistent. Would `ErrorCode.ERR_PredefinedTypeNotFound` be sufficient here, or should we use one diagnostic for all but improve the message?